src/WCSimWCSD.cc

```diff
@@ -91,19 +91,10 @@ G4bool WCSimWCSD::ProcessHits(G4Step* aStep, G4TouchableHistory*)
   G4float photonStartTime;
   G4ThreeVector photonStartPos;
   G4ThreeVector photonStartDir;
-  if (trackinfo) {
-    //Skip secondaries and match to mother process, eg. muon, decay particle, gamma from pi0/nCapture.
-    primParentID = trackinfo->GetPrimaryParentID();  //!= ParentID.
-    photonStartTime = trackinfo->GetPhotonStartTime();
-    photonStartPos = trackinfo->GetPhotonStartPos();
-    photonStartDir = trackinfo->GetPhotonStartDir();
-  }
-  else { // if there is no trackinfo, then it is a primary particle!
-    primParentID = aStep->GetTrack()->GetTrackID();
-    photonStartTime = aStep->GetTrack()->GetGlobalTime();
-    photonStartPos = aStep->GetTrack()->GetVertexPosition();
-    photonStartDir = aStep->GetTrack()->GetVertexMomentumDirection();
-  }
+  primParentID = aStep->GetTrack()->GetTrackID();
```

Comment:
In the original code, this will be the most upstream parent ID. Now it becomes the current track ID. How can we get the parent ID in the new code?

Reply:
Since this is meant as a seemless low-impact change, I'd recommend keeping the primary parent ID definiton the same as it currently is. But add the extra information of parent ID to everything through to and including WCSimRootCherenkovHit

Reply:
Ah, I'm sorry I forgot to also mention this 'fix'.

Actually, the current version does NOT save the primary parent in this variable. The track ID it does save is somewhat complicated... In the new verison it will take the direct parent if that parent track is saved. In the old version, it will be the same if the direct parent is a saved track and is anything other than a gamma (set [here](https://github.com/WCTE/WCSim/blob/32640318f1ef19fad8b15669242c4d63f6da65dd/src/WCSimTrackingAction.cc#L182-L186)), or if it is a gamma produced by a pi0 or nCapture or NeutronInelastic (set [here](https://github.com/WCTE/WCSim/blob/32640318f1ef19fad8b15669242c4d63f6da65dd/src/WCSimTrackingAction.cc#L180)).

So there are only two cases where anything is changed:

1. The choice of which true tracks are saved as WCSimRootTrack has been changed by the rest of this PR, and this can change whether that track would be the parent or not (because it never saves a parent ID of a track that isn't saved. This will change even if we kept the old code for the hit parent ID, since both old and new require the track to be saved.
2.  If the particle that produces a hit is itself produced by a gamma that is not produced by a pii0 or nCapture or NeutronInelastic, then the old code would save the ID of the gamma's parent instead of the gamme. The new code saves the ID of the gamma. I can't think of any situation where a gamma directly produces an optical photon that produces a hit, so this situation should never occur anyway.

The original reason (I believe, based on the comments) for the old version's logic is that certain electron tracks in an EM shower were not saved before, so the code was to be careful that it didn't save a parent ID of a gamma in the EM shower that was saved (unless it was one of these specific ones) but instead always traced back to a particle capable of producing Cherenkov light, even if that wasn't the particle that produced the light (because that particle's track wasn't saved.

So in the end, the only way to retain the old version's functionality would be to add back in all the arbitrary logic about which tracks to save and just use that to set the parent ID of the hit, which doesn't seem useful since we're now saving the other tracks anyway.

Reply:
Thanks for clearing up this complicated logic! I'm happy for the new code to do as you say

Just a request to make the variables/functions more descriptive of reality. So instead of "primary parent ID" or similar, something instead like "deepest saved track ID" (better options are potentially available. "Lowest" was my first thought, but doesn't work (the primary particles use the lowest numbers))

Reply:
Sure, I can do this later, about to board a flight... I had thought about renaming but just left it as it was, but agree it's better to change. Actually as it gets passed throughout the code, its variously called "primaryParentId" or "primParentId" or just "parentId". In the WCSimRootCherenkovHitTime class that users actually see, I think the method is simply "GetParentID".  I'd rather not change this (to avoid breaking anything downstream), so maybe to match that we just call it parentID in the internal variables too? Or if you want it to be very explicit, I'd go for something like savedParentTrackID or parentSavedTrackID since the one requirement is that the track is saved. What do you think?

Reply:
> savedParentTrackID or parentSavedTrackID

Yeah these are good. I like both

And I do think being explicit rather than adherring to old unclear convetions is better.
Since the only thing 95% of users will see is the method in `WCSimRootCherenkovHitTime`, then I think making the rest of the code use the new name would be good. And inside `WCSimRootCherenkovHitTime` maybe deprecate `GetParentID()` whilst adding e.g. `GetParentSavedTrackID()`

Enjoy your travels!